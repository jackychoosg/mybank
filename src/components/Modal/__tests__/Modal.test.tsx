import { render, cleanup, fireEvent } from '@testing-library/react'
import { Modal } from '../index'
describe('Modal component', () => {
  afterEach(cleanup)
  const onClose = jest.fn()
  const onConfirm = jest.fn()
  const title = 'This is title'
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <Modal onClose={onClose} onConfirm={onConfirm} title={title}>
        <div>Testing</div>
      </Modal>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should call onConfirm action', () => {
    const { getByTestId } = render(
      <Modal onClose={onClose} onConfirm={onConfirm} title={title}>
        <div>Testing</div>
      </Modal>
    )
    const confirmBtn = getByTestId('confirm')
    fireEvent.click(confirmBtn)
    expect(onConfirm).toBeCalled()
  })

  it('should call onCancel action', () => {
    const { getByTestId } = render(
      <Modal onClose={onClose} onConfirm={onConfirm} title={title}>
        <div>Testing</div>
      </Modal>
    )
    const cancelBtn = getByTestId('cancel')
    fireEvent.click(cancelBtn)
    expect(onClose).toBeCalled()
  })
})
