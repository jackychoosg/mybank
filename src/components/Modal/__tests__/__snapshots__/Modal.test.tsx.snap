// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-jSFjdj icVPtE"
  >
    <div
      class="sc-gKAaRy gRKpOc"
    >
      <span
        class="sc-iCoGMd jUdngs"
      />
      <div
        class="sc-pNWdM hEtHxe"
      >
        This is title
      </div>
      <div>
        Testing
      </div>
      <div
        class="sc-fujyAs bAckHs"
      >
        <button
          class="sc-hKFxyN cMgOKN"
          color="red"
          data-testid="cancel"
        >
          Cancel
        </button>
        <button
          class="sc-hKFxyN iZlfpJ"
          color="green"
          data-testid="confirm"
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
